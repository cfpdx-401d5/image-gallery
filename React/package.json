{
  "name": "image-gallery",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "body-parser": "^1.17.1",
    "cors": "^2.8.1",
    "express": "^4.15.2",
    "mongoose": "^4.8.7",
    "morgan": "^1.8.1",
    "react": "^15.4.2",
    "react-dom": "^15.4.2"
  },
  "devDependencies": {
    "chai": "^3.5.0",
    "chai-http": "^3.0.0",
    "enzyme": "^2.7.1",
    "enzyme-to-json": "^1.5.0",
    "eslint": "^3.17.1",
    "eslint-plugin-react": "^6.10.0",
    "mocha": "^3.2.0",
    "nodemon": "^1.11.0",
    "react-addons-test-utils": "^15.4.2",
    "react-scripts": "0.9.4"
  },
  "scripts": {
    "start": "react-scripts start",
    "start-server": "nodemon server.js",
    "build": "react-scripts build",
    "test": "react-scripts test --env=jsdom",
    "eject": "react-scripts eject"
  }
}
