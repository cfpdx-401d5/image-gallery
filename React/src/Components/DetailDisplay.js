import React from 'react';

export default function DetailDisplay(props) {
    return (
        <div>
            <h4>{props.babyBunBun.title}</h4>
            <a href={props.babyBunBun.url}>See It Here!</a>
            <p>{props.babyBunBun.description}</p>
            <button onClick={() => props.deleteBunny(props.babyBunIndex, props.babyBunBun._id)}>Delete Bunny</button>
        </div>
    )
}

DetailDisplay.propTypes = {
    deleteBunny: React.PropTypes.func,
    babyBunIndex: React.PropTypes.number,
    babyBunBun: React.PropTypes.object
};