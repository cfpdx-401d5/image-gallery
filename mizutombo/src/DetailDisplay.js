import React, { PropTypes } from 'react';

function DetailDisplay(props) {
  return (
    <div className='image-divs'>
      <p> Title: {props.image.title} </p>
      <p> Description: {props.image.description} </p>
      <a href={props.image.url}> Image Link </a>
      <button className='kill-button' onClick={() => props.deleteImage(props.image) }>
        Delete
      </button>
    </div>
  );
}

DetailDisplay.propTypes = {
  image: PropTypes.object.isRequired,
  deleteImage: PropTypes.func.isRequired,
};

export default DetailDisplay;
