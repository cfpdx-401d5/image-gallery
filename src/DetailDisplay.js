import React from 'react';

export default function DetailDisplay(props) {
    return (
        <div>
            <h4>{props.babyBunBun.title}</h4>
            <a href={props.babyBunBun.url}>See It Here!</a>
            <p>{props.babyBunBun.description}</p>
            <button onClick={() => props.deleteBunny(props.babyBunIndex)}>Delete Bunny</button>
        </div>
    )
}

DetailDisplay.propTypes = {
    deleteBunny: React.PropTypes.func,
    babyBunIndex: React.PropTypes.number,
    key: React.PropTypes.oneOfType([
        React.PropTypes.string,
        React.PropTypes.number,
    ]),
    babyBunBun: React.PropTypes.object
};