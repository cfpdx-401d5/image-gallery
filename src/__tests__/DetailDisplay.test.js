import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import DetailDisplay from '../DetailDisplay';

describe('DetailDisplay component: ', function () {

  it('renders the DetailDisplay component', () => {
    const rendered = shallow(<DetailDisplay image={{ title: '_Title', description: '_description', url: '_url' }} deleteImage={() => {} } />);
    expect(toJson(rendered)).toMatchSnapshot();
  });

});