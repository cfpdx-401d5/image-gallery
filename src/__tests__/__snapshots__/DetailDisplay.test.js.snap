exports[`DetailDisplay component:  renders the DetailDisplay component 1`] = `
<div
  className="image-divs">
  <p>
     Title: 
    _Title
     
  </p>
  <p>
     Description: 
    _description
     
  </p>
  <a
    href="_url">
     Image Link 
  </a>
  <button
    className="kill-button"
    onClick={[Function]}>
    Delete
  </button>
</div>
`;
